### PR TITLE
Remove separate .lintr_new

### PR DESCRIPTION
**Pull request by MichaelChirico**
Is there a reason to keep both workflows now @IndrajeetPatil ? Or is lint-changed-files sufficient?

Or maybe we should drop lint-changed-files and make the regular lint workflow error instead? My thinking is the latter will catch new lints that pop up in unchanged files as we improve linter logic.

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
> Or maybe we should drop lint-changed-files and make the regular lint workflow error instead? My thinking is the latter will catch new lints that pop up in unchanged files as we improve linter logic.

Yes, I would also personally prefer this option. 

E.g., in this very PR, `lint` is flagging three [new lints](https://github.com/r-lib/lintr/actions/runs/6188594718/job/16800944012?pr=2165) that we will discover only when the respective files change.

**Comment by MichaelChirico** [comment]
Some of those are fixed in #2152. I left this implicit assignment on purpose:

https://github.com/r-lib/lintr/blob/306a078d0da9f035e131ff62d9f4a432f20850ee/R/tree_utils.R#L9

I would mark it as `# nolint` pending #2015.

**Comment by MichaelChirico** [comment]
Marked as draft until #2152 is merged.

File: tests/testthat/test-methods.R [hunk 1]
```diff
@@ -144,7 +144,7 @@ test_that("split.lint works as intended", {
 test_that("within.list is dispatched", {
   l <- lint(text = "a=1\nb=2", linters = infix_spaces_linter())
   expect_silent({
-    l <- lapply(l, within, line_number <- line_number + 1L)
+    l <- lapply(l, within, line_number <- line_number + 1L) # nolint: implicit_assignment_linter. Workaround is pretty horrid.
```
Comment (IndrajeetPatil) [inline tests/testthat/test-methods.R:147]: This change is producing `line_length_linter()` lint.

**Review by IndrajeetPatil — APPROVED** [review]

**Pull request merged**